lib/core/text/ropes.nit

```diff
@@ -936,26 +936,26 @@ private class RopeSubstrings
 		pos += str.length
 		if pos > max then return
 		var it = iter.prev
-		var rnod = it.node
+		var rnod = it.as(not null).node
 		loop
 			if not rnod isa Concat then
-				it.ldone = true
-				it.rdone = true
+				it.as(not null).ldone = true
+				it.as(not null).rdone = true
 				str = rnod.as(FlatString)
 				iter = it.as(not null)
 				break
 			end
-			if not it.ldone then
+			if not it.as(not null).ldone then
 				rnod = rnod._left
-				it.ldone = true
+				it.as(not null).ldone = true
 				it = new RopeCharIteratorPiece(rnod, false, false, it)
-			else if not it.rdone then
-				it.rdone = true
+			else if not it.as(not null).rdone then
+				it.as(not null).rdone = true
 				rnod = rnod._right
 				it = new RopeCharIteratorPiece(rnod, false, false, it)
 			else
-				it = it.prev
-				rnod = it.node
+				it = it.as(not null).prev
+				rnod = it.as(not null).node
```

Comment:
Because with the current code, `it` must stay nullable since `it = it.prev`. It required a lot of code modification, and it was 3 a.m., I was not willing to do it. 

Now, at 6 p.m., I look at the name of the module and HELL NO, I'm not touching that code :p